internal/imei/imei.go

```diff
@@ -27,5 +26,34 @@ var (
 // Decode does NOT allocate under any condition. Additionally, it panics if b
 // isn't at least 15 bytes long.
 func Decode(b []byte) (code uint64, err error) {
-	panic(common.ErrNotImplemented)
+	if len(b) < 15 {
+		panic(ErrInvalid)
+	}
+
+	sum := 0
+	for i, d := range b {
+		if d < 0 || d > 9 {
```

Comment:
`d < '0' || d > '9'`?

Reply:
Hmm, thought imei is sent by bytes not symbols ;). so imei "490154203237518" in bytes []byte{4, 9, 0, 1, 5, 4, 2, 0, 3, 2, 3, 7, 5, 1, 8}. if it's symbols it's simplify things

Reply:
It's a string literal.